src/ophyd_async/core/_device/device_save_loader.py

```diff
@@ -0,0 +1,122 @@
+from enum import Enum
+from typing import Any, Dict, List
+
+import yaml
+from bluesky import Msg
+from numpy import ndarray
+
+from ophyd_async.core import Device, SignalRW
+
+
+def get_signal_RWs_from_device(
+    device: Device, path_prefix: str = "", signalRWs: Dict[str, SignalRW] = {}
+) -> Dict[str, SignalRW]:
+    """
+    Get all the SignalRWs from a device and store them with their dotted attribute
+    paths. Used by the save and load methods.
+
+    Parameters
+    ----------
+    device : Device
+        Ophyd device to retrieve read-write signals from.
+
+    path_prefix : str
+        For internal use, leave blank when calling the method.
+
+    SignalRWs : dict
+        A dictionary matching the string attribute path of a SignalRW with the
+        signal itself. Leave blank when calling the method.
+
+    Returns
+    -------
+    SignalRWs : dict
+        A dictionary matching the string attribute path of a SignalRW with the
+        signal itself.
+    """
+
+    for attr_name, attr in device.children():
+        dot = ""
+        # Place a dot inbetween the upper and lower class.
+        # Don't do this for highest level class.
+        if path_prefix:
+            dot = "."
+        dot_path = f"{path_prefix}{dot}{attr_name}"
+        if type(attr) is SignalRW:
+            signalRWs[dot_path] = attr
+        get_signal_RWs_from_device(attr, path_prefix=dot_path)
+    return signalRWs
+
+
+def save_device(device: Device, savename: str, ignore: List[str] = []):
```

Comment:
I would put this phase sorting and the ignore list in a device specific plan, and also move the yaml saving to another function:
```suggestion
def save_signals(signals: List[Dict[str, SignalRW]], ignore: Sequence[str] = ()) -> List[Dict[str, Any]]:
```